### ISSUE TITLE
Can we keep the same relay ports when remote port changed in media PASSTHRU mode? 

### ISSUE THREAD

**Issue by alexcme** [sipwise/rtpengine#119]
Related to issue #88 renegotiating ice/dtls-srtp mid-call, but, when we already have MEDIA_ISSET(media, PASSTHRU), do we really have to reset / create new relay when one of the sides changes port # only in re-INVITE? It actually breaks the old SBC functionality. We tested a quick fix disabling it when request (offer/answer) received over UDP control path and it looks ok.

**Comment by rfuchs** [comment]
Well, it may work in your case, but it would break other cases. The correct thing to do is to go along with what the client does and also change ports. Why is this a problem?

**Comment by alexcme** [comment]
Just want to point that it break an old "ng_" logic - we tried to replace rtpproxy with latest rtpengine and it did not work (again it only relevant to "legacy" UDP)

**Comment by rfuchs** [comment]
I don't understand. Behaviour should be the same between NG and UDP protocols.

**Comment by alexcme** [comment]
In "legacy" rtpproxy i.e. "PASSTHRU" mode, even when port # being changed on one of the sides it still keep the same relay, now we have it changed on the "other" side that make a trouble in our case.

**Comment by rfuchs** [comment]
Can you give a specific example?

**Comment by alexcme** [comment]
We'll collect traces from rtpproxy (legacy) and rtpengine to compare, but in short - kamailio as an SBC for SIP/TLS devices works fine with rtpproxy (legacy), but failing with rtpengine (UDP control) in hold/un-hold cases when held device (internal) changing port # (i.e. Music On Hold stream)

**Comment by rfuchs** [comment]
Well, rtpengine isn't designed to duplicate rtpproxy's behaviour in every aspect, if that's what you're getting at.

**Comment by alexcme** [comment]
I understand, but, again, do we really want to create new relay in PASSTHRU mode? All crypto's contex set to NULL, so what's the point?

**Comment by rfuchs** [comment]
It's not about crypto nor about passthrough. It's honouring the client's choice. The client switched ports and probably for a reason, thus so should we.

**Comment by alexcme** [comment]
Understandable, but kind of hard to explain to ISP providers, such as AT&T :(

**Comment by rfuchs** [comment]
You're welcome to submit a patch to make this behaviour optional, even though I still don't understand what the problem with switching ports is.

**Comment by alexcme** [comment]
I'll fork it out for review

**Comment by gled-rs** [comment]
Rfuchs, 

Jumping on it as I saw this bug, I saw some pstn gateways ( won't cite any brands, but those who had DTMF issues in the past too, yeah you know ;) ) behaving very badly when being reinvited ( up to 5s of blank audio, etc... ) with media changes, no issues though when the reinvite is just a session reinvite. I had to put SBCs in front of those to ensure there would be no media change, and no media interruption at all.. 

Maybe it's your case Alexcme ? 

An optional behaviour change is IMHO recommended, as this is a corner case, and a workaround of some buggy implementation. The correct behaviour implemented today by rtpengine is important to conserve for those of us using it successfully.

My 2 cents...
